src/server/replica.cc

```diff
@@ -173,7 +178,7 @@ void Replica::MainReplicationFb() {
   SetShardStates(true);
 
   error_code ec;
-  while (state_mask_.load() & R_ENABLED) {
+  while (!cntx_.IsCancelled() && state_mask_.load() & R_ENABLED) {
```

Comment:
How do retries work then? 🤨 Will this ever be true after an error?

Reply:
we should check context for cancel state when we stop replica

Reply:
And if we just want to retry connecting? The context being cancelled is not a citeria

Reply:
why we should retry if context is cancelled?